docs/book/src/developer/logging.md

```diff
@@ -0,0 +1,158 @@
+# Logging
+The Cluster API project is committed to improving the SRE/developer experience when troubleshooting issues, and logging
+plays an important part in this goal.
+
+In Cluster API we strive to follow three principles while implementing logging:
+
+- **Logs are for SRE & developers, not for users!**
+  Whenever an end user is required to read logs to understand what is happening in the system, most probably there is an
+  opportunity for improvement of other observability in our API, like e.g. conditions and events.
+- **Navigating logs should be easy**:
+  We should make sure that SREs/Developers can easily drill down logs while investigating issues, e.g. by allowing to
+  search all the log entries for a specific Machine object, eventually across different controllers/reconciler logs.
+- **Cluster API developers MUST use logs!**
+  As Cluster API contributors you are not only the ones that implement logs, but also the first users of them. Use it!
+  Provide feedback!
+
+## Upstream Alignment
+
+Kubernetes defines a set of [logging conventions](https://git.k8s.io/community/contributors/devel/sig-instrumentation/logging.md), 
+as well as tools and libraries for logging.
+
+As a CAPI project we must benefit from this great work, ensuring consistency with the ecosystem and focusing our
+resources on what is specific to this project.
+
+## Continuous improvement
+
+While we all would like to have top-notch logs, we should recognize the fact that we already have a huge amount of code,
+and changing logs is not the kind of work that attracts many contributors.
+
+As a consequence we aim for continuous improvement, starting from a set of foundational items, that will enable a long
+tail of small improvements over time.
```

Comment:
Remove these? Let's get right to the point of defining our logging

Reply:
done